contracts/IdleCDOCardManager.sol

```diff
@@ -100,25 +125,69 @@ contract IdleCDOCardManager is ERC721Enumerable {
     return ratioAA.add(ratioBB);
   }
 
-  function balance(uint256 _tokenId) public view returns (uint256 balanceAA, uint256 balanceBB) {
-    Card memory pos = card(_tokenId);
-    require(pos.cardAddress != address(0), "inexistent card");
+  function cardIndexes(uint256 _tokenId) public view returns (uint256[] memory _cardIndexes) {
+    return _cards[_tokenId];
+  }
+
+  function balance(uint256 _tokenId, uint256 _index) public view returns (uint256 balanceAA, uint256 balanceBB) {
+    require(_isCardExists(_tokenId, _index), "inexistent card");
+    Card memory pos = card(_tokenId, _index);
     IdleCDOCard _card = IdleCDOCard(pos.cardAddress);
-    return _card.balance();
+    return cardBalance(pos.idleCDOAddress,pos.cardAddress);
   }
 
   function percentage(uint256 _percentage, uint256 _amount) private pure returns (uint256) {
-    require(_percentage < RATIO_PRECISION.add(1), "percentage should be between 0 and 1");
+    require(_percentage < RATIO_PRECISION.add(1), "% should be between 0 and 1");
     return _amount.mul(_percentage).div(RATIO_PRECISION);
   }
 
-  function _mint() private returns (uint256) {
-    _tokenIds.increment();
+  function _isCardExists(uint256 _tokenId, uint256 _index) internal view virtual returns (bool) {
+    return _cards[_tokenId].length != 0 && _cards[_tokenId].length > _index;
+  }
+
+  function _depositToCard(
+    IdleCDOCard _card,
+    address _idleCDOAddress,
+    uint256 _risk,
+    uint256 _amount
+  ) private {
+    // check if _idleCDOAddress exists in idleCDOAddress array
+    require(isIdleCDOListed(_idleCDOAddress), "IdleCDO address is not listed");
+
+    IERC20Detailed underlying = IERC20Detailed(IdleCDO(_idleCDOAddress).token());
+
+    // transfer amount to cards protocol
+    underlying.safeTransferFrom(msg.sender, address(_card), _amount);
 
-    uint256 newItemId = _tokenIds.current();
-    _mint(msg.sender, newItemId);
+    // calculate the amount to deposit in BB
+    // proportional to risk
+    uint256 depositBB = percentage(_risk, _amount);
 
-    return newItemId;
+    // calculate the amount to deposit in AA
+    // inversely proportional to risk
+    uint256 depositAA = _amount.sub(depositBB);
```

Comment:
you can avoid declaring this variable and use the result directly in the `mint` call

Reply:
done